cpp/src/parquet/arrow/reader.cc

```diff
@@ -494,24 +494,30 @@ Status FileReader::Impl::ReadRowGroup(int row_group_index,
 
   auto rg_metadata = reader_->metadata()->RowGroup(row_group_index);
 
-  int num_columns = static_cast<int>(indices.size());
-  std::vector<std::shared_ptr<Column>> columns(num_columns);
+  // We only need to read schema fields which have columns indicated
+  // in the indices vector
+  std::vector<int> field_indices;
+  if (!ColumnIndicesToFieldIndices(*reader_->metadata()->schema(), indices,
+                                   &field_indices)) {
+    return Status::Invalid("Invalid column index");
+  }
+  int num_fields = static_cast<int>(field_indices.size());
```

Comment:
IIUC, the fix is that `field_indices.size()` can be different from `indices.size()`?

Reply:
Yes, this is the same as we do in https://github.com/apache/arrow/blob/6c4118b274cadf044b2d0581401a018f0a438205/cpp/src/parquet/arrow/reader.cc#L548-L556

That fix there was not applied to the single RowGroup reads.